x-pack/legacy/plugins/infra/common/http_api/logs/summary.ts

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import * as rt from 'io-ts';
+
+export const LOGS_SUMMARY_PATH = '/api/logs/summary';
+export const LOGS_SUMMARY_HIGHLIGHTS_PATH = '/api/logs/summary/highlights';
+
+// const esDate = rt.union([rt.string, rt.number]);
+
+export const logsSummaryRequestRT = rt.type({
+  startDate: rt.number,
+  endDate: rt.number,
+  bucketSize: rt.number,
+  query: rt.union([rt.string, rt.undefined, rt.null]),
+});
+
+export const logsSummaryHighlightsRequestRT = rt.intersection([
+  logsSummaryRequestRT,
+  rt.type({
+    highlightTerms: rt.array(rt.string),
+  }),
+]);
+
+export type LogsSummaryRequest = rt.TypeOf<typeof logsSummaryRequestRT>;
+export type LogsSummaryHighlightsRequest = rt.TypeOf<typeof logsSummaryHighlightsRequestRT>;
+
+export const logsSummaryResponseRT = rt.type({
```

Comment:
We could choose to wrap the responses in an object with a `data` key so we can add additional metadata to the response later on (such as timing/debug information or HATEOAS links). We already did that in the other logs-related HTTP responses. Do you think that's reasonable?